### PR TITLE
[WIP] x86: add kernel 5.10 support

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Bump kernel.

### CODE REVIEW

File: package/kernel/linux/modules/netdevices.mk [hunk 1]
```diff
@@ -539,15 +539,15 @@ endef
 
 $(eval $(call KernelPackage,8139cp))
 
-
```
Comment (adschm) [inline package/kernel/linux/modules/netdevices.mk:542]: That should be kept.

File: package/kernel/linux/modules/netdevices.mk [hunk 2]
```diff
@@ -539,15 +539,15 @@ endef
...
 define KernelPackage/r8169
   SUBMENU:=$(NETWORK_DEVICES_MENU)
   TITLE:=RealTek RTL-8169 PCI Gigabit Ethernet Adapter kernel support
   DEPENDS:=@PCI_SUPPORT +kmod-mii +r8169-firmware +kmod-phy-realtek
   KCONFIG:=CONFIG_R8169 \
     CONFIG_R8169_NAPI=y \
     CONFIG_R8169_VLAN=n
-  FILES:=$(LINUX_DIR)/drivers/net/ethernet/realtek/r8169.ko
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/realtek/r8169.ko \
+	  $(LINUX_DIR)/drivers/net/phy/mdio_devres.ko
```
Comment (adschm) [inline package/kernel/linux/modules/netdevices.mk:550]: Please make hanging indent consistent, at least locally.
Shouldn't this get a kernel switch?

File: target/linux/x86/patches-5.10/011-tune_lzma_options.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
+Index: linux-5.10.16/scripts/Makefile.lib
+===================================================================
```
Comment (adschm) [inline target/linux/x86/patches-5.10/011-tune_lzma_options.patch:2]: This does not look like you refreshed patches?

**Comment by adschm** [comment]
Note that it's easier to read and review if you first copy stuff and then apply the changes in separate commits, so one sees the diff directly.

**Comment by PolynomialDivision** [comment]
> Note that it's easier to read and review if you first copy stuff and then apply the changes in separate commits, so one sees the diff directly.

But then I have a commit where x86 is not compiling anymore? If that is okay, I can change it. :)

File: target/linux/x86/config-5.10 [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+CONFIG_GENERIC_PCI_IOMAP=y
+CONFIG_GENERIC_SMP_IDLE_THREAD=y
+CONFIG_GENERIC_STRNCPY_FROM_USER=y
+CONFIG_GENERIC_STRNLEN_USER=y
+CONFIG_GENERIC_TIME_VSYSCALL=y
+CONFIG_GENERIC_VDSO_32=y
+# CONFIG_GEOS is not set
+CONFIG_GLOB=y
+# CONFIG_HABANA_AI is not set
+# CONFIG_HANGCHECK_TIMER is not set
```
Comment (abajk) [inline target/linux/x86/config-5.10:191]: Config files should be refreshed. I see this file is not sorted. For example, CONFIG_HANGCHECK_TIMER and CONFIG_CPU_IDLE_GOV_LADDER are in the wrong order.

**Comment by curtdept** [comment]
@PolynomialDivision I have a dupe PR to yours with a few updates based on some of the comments above, did you want to make revs and ill close mine, or do you want to close yours?

File: package/kernel/linux/modules/netdevices.mk (hunk 2 above)
Comment (curtdept) [inline package/kernel/linux/modules/netdevices.mk:550]: please modify to target specific kernel ie . @ge5.10

File: target/linux/x86/patches-5.10/011-tune_lzma_options.patch [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+--- linux-5.10.16.orig/scripts/Makefile.lib
++++ linux-5.10.16/scripts/Makefile.lib
```
Comment (curtdept) [inline target/linux/x86/patches-5.10/011-tune_lzma_options.patch:4]: Please ensure file name are consistent with other patches ie a / b style

ex:
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib

File: target/linux/x86/patches-5.10/011-tune_lzma_options.patch [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ # ---------------------------------------------------------------------------
+ 
+ quiet_cmd_lzma = LZMA    $@
+-      cmd_lzma = { cat $(real-prereqs) | $(LZMA) e -d20 -lc1 -lp2 -pb2 -eos -si -so; $(size_append); } > $@
++      cmd_lzma = { cat $(real-prereqs) | $(LZMA) e -lc8 -eos -si -so; $(size_append); } > $@
+ 
+ quiet_cmd_lzo = LZO     $@
+       cmd_lzo = { cat $(real-prereqs) | $(KLZOP) -9; $(size_append); } > $@
+Index: linux-5.10.16/arch/x86/include/asm/boot.h
+===================================================================
```
Comment (curtdept) [inline target/linux/x86/patches-5.10/011-tune_lzma_options.patch:15]: please remove these quilt edit index references to be consistent with other patches

**Review by curtdept — CHANGES_REQUESTED** [review]
- Where did patch target/linux/x86/patches-5.4/300-pcengines_apu1_led.patch go?
- Please copy configs for all subtargets ie 64, generic, etc

File: target/linux/x86/patches-5.10/011-tune_lzma_options.patch (hunk 5 above)
Comment (adschm) [reply target/linux/x86/patches-5.10/011-tune_lzma_options.patch:4]: Just refresh the patches aber adding them, the script will do this automatically.

File: target/linux/x86/patches-5.10/011-tune_lzma_options.patch (hunk 6 above)
Comment (adschm) [reply target/linux/x86/patches-5.10/011-tune_lzma_options.patch:15]: see above ...

**Comment by guidosarducci** [comment]
@PolynomialDivision @curtdept @adschm
I needed to test some of my packages against x86 on 5.10, and ended up completing the x86/5.10 update work in order to make progress. If it can help move this PR forward, you could have a look at https://github.com/guidosarducci/openwrt/commits/master-add-x86-5.10. I'll refrain from muddying the waters with yet a third PR.

**Comment by PolynomialDivision** [comment]
> @PolynomialDivision @curtdept @adschm
> I needed to test some of my packages against x86 on 5.10, and ended up completing the x86/5.10 update work in order to make progress. If it can help move this PR forward, you could have a look at https://github.com/guidosarducci/openwrt/commits/master-add-x86-5.10. However, I'll refrain from muddying the waters with yet a third PR. Also note the last 4 commits should be eventually squashed.

Should I maybe just close for your PR? Sry, I was very very busy the last days.

**Comment by guidosarducci** [comment]
I posted hoping someone could use that branch, but I don't have a PR and am reluctant to make a third one when it's difficult to get review time anyway. For example, @adschm has already looked through 2 PRs doing the same thing; who would want to look through a third? If you would like me to PR my branch then I will, I just didn't want to muddy waters further as I said.

@PolynomialDivision @curtdept Let me know what you guys prefer, and if you have time please build my branch to confirm it works for you. I've already done a fair bit of review and testing so I expect it should be fine.

@adschm Please have some patience for this game of musical PRs, as it should be over soon. If you have a preference for direction let us know too.

**Comment by curtdept** [comment]
TBH I'm less and less caring about who makes a PR that wins, just that one does, x86 builds represent a fairly significant user base and would be good for package and kernel testing 5.10. @PolynomialDivision can you update this one and keep it going?

**Comment by adschm** [comment]
Well, I won't merge an x86 PR anyway. I'm not active on this platform.

The main point is that closing and opening a new PR will remove all the previous discussion from a reviewers eyes. This might not be that harmful if it's become irrelevant anyway; but if discussion is needed to understand the current state or provide other additional info, that's lost and needs to be supplied again (obviously).

So, essentially you should evaluate whether you want to pursue the issue afterwards ...

**Comment by guidosarducci** [comment]
AFAICS there's no relevant technical context that needs preserving in either this or the other closed PR. A good set of commits should provide such context in their descriptions in any case.

@PolynomialDivision Next steps are really up to you, in choosing either to update/fix this PR (taking whatever you like from [my branch](https://github.com/guidosarducci/openwrt/commits/master-add-x86-5.10)) or just closing it.

**Comment by PolynomialDivision** [comment]
I closed. I can not evaluate on a x86 board right now. :/ Sry.

**Comment by guidosarducci** [comment]
OK, fine if you prefer closing. I'll PR the branch I referred to and we'll take it from there.

I've tested virtually, using QEMU with x86/64, x86/legacy and x86/generic. The x86/geode subtarget needs physical HW, but I've at least compile-tested that.

**Pull request closed without merging**